Added support for Wifi 802.11a (5 GHz)

ftoure67 opened the pull request:
Related Issue: This PR fixes/closes {2990, 2914}

Replaced the Wifi channel grid by a combo list of channels. The code is still present but the grid is hidden.
Added the display of the country code.

Screenshots: https://photos.google.com/search/_tra_/photo/AF1QipMJpEdDwJjAGeHWhWkzQKhKRCJoh1ppSDObGawv

Signed-off-by: ftoure67 <famory.toure@gmail.com>

MMaiero commented:
Hi @ftoure67 
please have a look at the sonarcloud results that are failing the build: https://sonarcloud.io/dashboard?id=org.eclipse.kura%3Akura&branch=PR-3329
Is this PR taking place of the previous? If so, can the previous be closed?

ftoure67 commented:
Hi Matteo,

Yes. I'm checking the sonarcloud errors.
This PR replaces the first one. I have squashed all the commits in a fresh
branch to make it simpler.

Regards,

Famory TOURE

Le mer. 17 mars 2021 à 08:42, Matteo Maiero ***@***.***> a
écrit :

> Hi @ftoure67 <https://github.com/ftoure67>
> please have a look at the sonarcloud results that are failing the build:
> https://sonarcloud.io/dashboard?id=org.eclipse.kura%3Akura&branch=PR-3329
> Is this PR taking place of the previous? If so, can the previous be closed?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/kura/pull/3329#issuecomment-800869144>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AECCNVBJMFAC2NSM7EITB23TEBMNLANCNFSM4ZISM3NA>
> .
>

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/NetworkAdminService.java:
```diff
@@ -213,4 +214,26 @@ public void setFirewallPortForwardingConfiguration(
+     *            - name of WiFi interface
+     * @return list of channels and frequencies.
+     * @throws KuraException
+     * @since 1.2
```
This should be 2.2 since that is the bundle version

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/NetworkAdminService.java:
```diff
@@ -213,4 +214,26 @@ public void setFirewallPortForwardingConfiguration(
+     * 
+     * @return Name of the Country Code or 00 if unknown.
+     * @throws KuraException
+     * @since 1.2
```
This should be 2.2 since that is the bundle version

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:
```diff
@@ -0,0 +1,82 @@
+
+import org.osgi.annotation.versioning.ProviderType;
+
+@ProviderType
```
Misses the since 2.2 annotation

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:
```diff
@@ -0,0 +1,82 @@
+import org.osgi.annotation.versioning.ProviderType;
+
+@ProviderType
+public class WifiChannel {
```
Is it possible to add the Javadoc?

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiConfig.java:
```diff
@@ -200,6 +209,14 @@ public void setIgnoreSSID(boolean ignoreSSID) {
         this.ignoreSSID = ignoreSSID;
     }
 
+    public String getWifiCountryCode() {
```
Needs the since 2.2 annotation

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiConfig.java:
```diff
@@ -200,6 +209,14 @@ public void setIgnoreSSID(boolean ignoreSSID) {
+        return wifiCountryCode;
+    }
+
+    public void setWifiCountryCode(String wifiCountryCode) {
```
Needs the since 2.2 annotation

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiConfig.java:
```diff
@@ -310,6 +328,14 @@ public boolean isValid() {
         return this.mode != null ? true : false;
     }
 
+    public List<WifiChannel> getChannelFrequencies() {
```
Needs the since 2.2 annotation

ftoure67 replied:
OK

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiConfig.java:
```diff
@@ -310,6 +328,14 @@ public boolean isValid() {
+        return channelFrequencies;
+    }
+
+    public void setChannelFrequencies(List<WifiChannel> channelFrequencies) {
```
Needs the since 2.2 annotation

ftoure67 replied:
OK

MMaiero commented on kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF:
```diff
@@ -16,6 +16,7 @@ Import-Package: javax.microedition.io,
  org.apache.commons.net.util;version="3.1.0",
  org.eclipse.kura;version="[1.1,2.0)",
  org.eclipse.kura.comm;version="[1.0,2.0)",
+ org.eclipse.kura.command;version="[1.2,1.3)",
```
I believe this range should be [1.2,2.0) since we are using the command API, not implementing them

ftoure67 replied:
OK

MMaiero commented on kura/org.eclipse.kura.linux.net/META-INF/MANIFEST.MF:
```diff
@@ -28,7 +29,7 @@ Import-Package: javax.microedition.io,
  org.eclipse.kura.net.modem;version="[2.1,2.2)",
  org.eclipse.kura.net.route;version="[1.0,1.1)",
- org.eclipse.kura.net.wifi;version="[2.0,3.0)",
+ org.eclipse.kura.net.wifi;version="[2.1,2.2)",
```
Why are we restricting the range here?

MMaiero commented on kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/util/IwListChannelTool.java:
```diff
@@ -0,0 +1,109 @@
+import org.slf4j.LoggerFactory;
+import org.eclipse.kura.net.wifi.WifiChannel;
+
+public class IwListChannelTool extends IwCapabilityTool {
```
Since we are not using IwList but Iw this class still needs to extend IwCapabilityTool? Or those methods could be added to the extended class?

MMaiero commented on kura/org.eclipse.kura.net.admin/META-INF/MANIFEST.MF:
```diff
@@ -14,7 +14,7 @@ Import-Package: javax.microedition.io,
  org.eclipse.kura.core.linux.executor;version="[1.0,2.0)",
  org.eclipse.kura.core.linux.util;version="[1.0,2.0)",
- org.eclipse.kura.core.net;version="[1.0,2.0)",
+ org.eclipse.kura.core.net;version="[1.0,2.1)",
```
If the net APIs have moved to 2.x I believe this range needs to be changed to [2.0,3.0). If not, the range can stay the same, right?

ftoure67 replied:
It probably comes from a merge. I will move it back to 2.0.

MMaiero commented on kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/NetworkAdminServiceImpl.java:
```diff
@@ -1228,6 +1230,17 @@ public void setFirewallNatConfiguration(List<FirewallNatConfig> natConfigs) thro
+    @Override
+    public List<WifiChannel> getWifiFrequencies(String ifaceName) throws KuraException {
+        List<WifiChannel> frequencies = IwListChannelTool.probeChannels(ifaceName, this.executorService);
+        return frequencies;
```
maybe this can be simply: `return IwListChannelTool.probeChannels(ifaceName, this.executorService);`

ftoure67 replied:
OK.

MMaiero commented on kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/visitor/linux/HostapdConfigWriter.java:
```diff
@@ -316,11 +316,11 @@ private String updateInterfaceAndDriver(String interfaceName, String interfaceDr
 
     private String updateChannels(WifiConfig wifiConfig, String fileAsString) throws KuraException {
-        if (wifiConfig.getChannels()[0] > 0 && wifiConfig.getChannels()[0] <= 14) {
+        if (wifiConfig.getChannels()[0] > 0 && wifiConfig.getChannels()[0] <= 200) {
```
If as reported in the message 124 is the maximum value, why are we setting here 200?

ftoure67 replied:
Yes. The limit is in fact 165. I will also change the message.

MMaiero commented on kura/org.eclipse.kura.web2/META-INF/MANIFEST.MF:
```diff
@@ -55,11 +55,11 @@ Import-Package: com.eclipsesource.json;version="0.9.5",
- org.eclipse.kura.net.modem;version="[2.1,3.0)",
- org.eclipse.kura.net.wifi;version="[2.0,3.0)",
+ org.eclipse.kura.net.modem;version="[2.0,3.0)",
+ org.eclipse.kura.net.wifi;version="[2.1,2.2)",
```
Why the range here is strict? I believe the UI is not implementing but only using the wifi APIs.

ftoure67 replied:
OK. I will extend to 3.0.

MMaiero commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:
```diff
@@ -181,6 +181,10 @@
     FormLabel labelPing;
     @UiField
     FormLabel labelIgnore;
+    @UiField
```
Add Sterwen to the copyright headers as contributor

ftoure67 replied:
OK

MMaiero commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/net/wifi/WifiChannel.java:
```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
```
I believe that the correct copyright header for the newly created files that you are contributing is something like:
```
/*******************************************************************************
  * Copyright (c) 2021 Sterwen-Technology and others
  * 
  * This program and the accompanying materials are made
  * available under the terms of the Eclipse Public License 2.0
  * which is available at https://www.eclipse.org/legal/epl-2.0/
  * 
  * SPDX-License-Identifier: EPL-2.0
  * 
  * Contributors:
  *  Sterwen-Technology
  ******************************************************************************/
```

MMaiero commented:
I believe you should rebase and not merge develop changes. Otherwise the eca check will complain

ftoure67 committed:
Added support for Wifi 802.11a (5 GHz)

Related Issue: This PR fixes/closes {2990, 2914}

Replaced the Wifi channel grid by a combo list of channels. The code is still present but the grid is hidden.
Added the display of the country code.

Screenshots: https://photos.google.com/search/_tra_/photo/AF1QipMJpEdDwJjAGeHWhWkzQKhKRCJoh1ppSDObGawv

Signed-off-by: ftoure67 <famory.toure@gmail.com>

MMaiero commented:
Hi @ftoure67 
it seems that the build is failing in the tests:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:testCompile (compiletests) on project org.eclipse.kura.linux.net.test: Compilation failure: Compilation failure:
[ERROR] /Users/matteo.maiero/git_eclipse/kura/kura/test/org.eclipse.kura.linux.net.test/src/test/java/org/eclipse/kura/linux/net/util/IwCapabilityToolTest.java:[193,25] cannot find symbol
[ERROR]   symbol:   variable IwListChannelTool
[ERROR]   location: class org.eclipse.kura.linux.net.util.IwCapabilityToolTest
[ERROR] /Users/matteo.maiero/git_eclipse/kura/kura/test/org.eclipse.kura.linux.net.test/src/test/java/org/eclipse/kura/linux/net/util/IwCapabilityToolTest.java:[222,30] cannot find symbol
[ERROR]   symbol:   variable IwListChannelTool
[ERROR]   location: class org.eclipse.kura.linux.net.util.IwCapabilityToolTest
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :org.eclipse.kura.linux.net.test
```

ftoure67 commented:
Hi Matteo,

OK. I have made the change. It was passing on my machine because the class
was probably still
in my local maven repository.

Regards,

Famory

Le mer. 24 mars 2021 à 06:55, Matteo Maiero ***@***.***> a
écrit :

> Hi @ftoure67 <https://github.com/ftoure67>
> it seems that the build is failing in the tests:
>
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:testCompile (compiletests) on project org.eclipse.kura.linux.net.test: Compilation failure: Compilation failure:
> [ERROR] /Users/matteo.maiero/git_eclipse/kura/kura/test/org.eclipse.kura.linux.net.test/src/test/java/org/eclipse/kura/linux/net/util/IwCapabilityToolTest.java:[193,25] cannot find symbol
> [ERROR]   symbol:   variable IwListChannelTool
> [ERROR]   location: class org.eclipse.kura.linux.net.util.IwCapabilityToolTest
> [ERROR] /Users/matteo.maiero/git_eclipse/kura/kura/test/org.eclipse.kura.linux.net.test/src/test/java/org/eclipse/kura/linux/net/util/IwCapabilityToolTest.java:[222,30] cannot find symbol
> [ERROR]   symbol:   variable IwListChannelTool
> [ERROR]   location: class org.eclipse.kura.linux.net.util.IwCapabilityToolTest
> [ERROR] -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
> [ERROR]
> [ERROR] After correcting the problems, you can resume the build with the command
> [ERROR]   mvn <args> -rf :org.eclipse.kura.linux.net.test
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/kura/pull/3329#issuecomment-805522532>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AECCNVAMG7LW6TFYZ44V56TTFF5EXANCNFSM4ZISM3NA>
> .
>
